msal4j-sdk/src/test/java/com/microsoft/aad/msal4j/AcquireTokenSilentlyTest.java

```diff
@@ -45,7 +56,71 @@ void confidentialAppAcquireTokenSilently_emptyCache_MsalClientException() throws
 
         ExecutionException ex = assertThrows(ExecutionException.class, future::get);
 
-        assertTrue(ex.getCause() instanceof MsalClientException);
+        assertInstanceOf(MsalClientException.class, ex.getCause());
         assertTrue(ex.getMessage().contains(AuthenticationErrorMessage.NO_TOKEN_IN_CACHE));
     }
+
+    @Test
+    void publicAppAcquireTokenSilently_claimsSkipCache() throws Throwable {
+
+        PublicClientApplication application = PublicClientApplication.builder("client_id")
+                .instanceDiscovery(false)
+                .authority("https://some.authority.com/realm")
+                .build();
+
+        application.tokenCache.deserialize(cache);
+
+        SilentParameters parameters = SilentParameters.builder(Collections.singleton("scopes"), basicAccount).build();
+
+        IAuthenticationResult result = application.acquireTokenSilently(parameters).get();
+
+        //Confirm cached dummy token returned from silent request
+        assertNotNull(result);
+        assertEquals("token", result.accessToken());
+
+        ClaimsRequest cr = new ClaimsRequest();
+        cr.requestClaimInAccessToken("something", null);
+
+        parameters = SilentParameters.builder(Collections.singleton("scopes"), basicAccount).claims(cr).build();
+        CompletableFuture<IAuthenticationResult> future = application.acquireTokenSilently(parameters);
+
+        //Confirm cached dummy token ignored when claims are part of request
+        ExecutionException ex = assertThrows(ExecutionException.class, future::get);
```

Comment:
These asserts are strange. Would it make sense to just merge the improvment where you expose "TokenSource" and assert on that?